libs/api/domains/driving-license/src/lib/drivingLicense.service.ts

```diff
@@ -243,20 +243,26 @@ export class DrivingLicenseService {
   async getQualityPhoto(
     nationalId: User['nationalId'],
   ): Promise<QualityPhotoResult> {
-    const hasQualityPhoto = await this.drivingLicenseApi.getHasQualityPhoto({
-      nationalId,
-    })
-    if (!hasQualityPhoto) {
+    let hasQualityPhoto
+
+    try {
+      hasQualityPhoto = await this.drivingLicenseApi.getHasQualityPhoto({
+        nationalId,
+      })
+    } catch (e) {
```

Comment:
Best practice in error handling is to only catch and handle “expected” issues which we know how to handle, or the User should know about in a special way.

It looks like in this case the only “expected” issue is if there is no quality photo stored (404). In GraphQL land, that could just be represented as `hasQualityPhoto: false`.

If there’s more than one expected issue that can occur, it’s best to have them in an Enum, so the UI as more flexibility in how to display them, eg with translated messages. Only if there is a big variability in “expected” issues which we can’t know about in GraphQL land should we allow some kind of string errorMessage from the external API. But a string “errorMessage” is generally bad since we can’t easily localize it.

ALL other (“unexpected”) errors should **not** be handled in services/resolvers. This error could be anything from a JS syntax error to network issues, authentication issues and service issues. We should never swallow it, hide it or try to recover from it. Instead we need to log these issue and handle them consistently (globally, not in each and every UI/Resolver/Service).

If you catch an error which you don’t expect, you should re-throw it. Then a global API interceptor can log it in a consistent way and send it to the client where it can also be handled in a consistent way (toast or error in external data provider). The ops team will notice the errors in the logs so they can be addressed.

Tldr, I suggest getting rid of errorMessage and just return `hasQualityPhoto: true/false`. Below the 404 case you should just `throw e` again.